gateway/handler.go

```diff
@@ -640,6 +648,39 @@ func customResponseFormat(r *http.Request) (mediaType string, params map[string]
 	return "", nil, nil
 }
 
+// Add 'Content-Location' headers for non-default response formats. This allows
+// correct caching of such format requests when the format is passed via the
+// Accept header, for example.
+func addContentLocation(r *http.Request, w http.ResponseWriter, rq *requestData) {
+	if rq.responseFormat == "" {
+		return
+	}
```

Comment:
I think we should also skip if `format` param is already present.

`Content-Location` should be returned only when `format` is missing.

```suggestion
  // Skip generating Content-Location if 
  // no explicit format was requested via Accept HTTP header,
  // or if it was requested via URL query parameter.
	if rq.responseFormat == "" || r.URL.Query().Get("format") != "" {
		return
	}
```

Rationale:
If we return `Content-Location` when `format` is already present in the original request, we will be creating a surface for cache-miss on subsequent requests due to `Content-Location` having query parameters in different order than the request made by non-golang client (that may have them in different order).

Cloudflare has "Enable Query String Sort" feature but it is only on Enterprise plan, so would not be something others can use.